scripts/import/import_clinvar_xml

```diff
@@ -1441,7 +1443,23 @@ sub clean_old_data{
   my $phenfeat_del_sth   = $dba->dbc->do(qq[ delete from phenotype_feature where source_id in (select source_id from source where name ='ClinVar') ]);
 
   # remove previously inserted variation from ClinVar + associated records
-  my @tables = qw/variation_synonym failed_variation variation_set_variation allele_synonym/;
+  # remove from allele_synonym
+  # first check which entries to remove
+  my $var_syn_sel_sth    = $dba->dbc()->prepare(qq[ select allele_synonym_id from allele_synonym where variation_id in
+                                                    (select variation_id from variation v, source s
+                                                    where v.source_id = s.source_id and s.name ='ClinVar')
+                                                ]);
+  $var_syn_sel_sth->execute();
+  my $data = $var_syn_sel_sth->fetchall_arrayref();
+  # remove entries from allele_synonym
+  foreach my $allele_syn (@{$data}){
+    my $allele_syn_id = $allele_syn->[0];
+    my $allele_syn_delete_sth = $dba->dbc()->prepare(qq[ delete from allele_synonym where allele_synonym_id = $allele_syn_id ]);
+    $allele_syn_delete_sth->execute();
```

Comment:
It would be good to capture fails and die here, though the same goes for all the clean up steps.